src/ucp/core/ucp_context.h

```diff
@@ -302,13 +304,22 @@ typedef struct ucp_tl_iface_atomic_flags {
 #define UCP_CONTEXT_CHECK_FEATURE_FLAGS(_context, _flags, _action) \
     do { \
         if (ENABLE_PARAMS_CHECK && \
-            ucs_unlikely(!((_context)->config.features & (_flags)))) { \
+            ucs_unlikely(!((_context)->config.features & (_flags)))) {  \
             size_t feature_list_str_max = 512; \
-            char *feature_list_str = ucs_alloca(feature_list_str_max); \
+            char *feature_list_str = ucs_alloca(feature_list_str_max);  \
+            const char *ucp_feature_str[] = { \
```

Comment:
this would define and initialize a local array in every place UCP_CONTEXT_CHECK_FEATURE_FLAGS macro is instantiated
it should be global array
(only in case of rdmacm ep it could be defined inside the function)